src/taipy/core/_version/_utils.py

```diff
@@ -22,3 +25,11 @@ def _version_migration() -> str:
     else:
         _VersionManager._get_or_create("LEGACY-VERSION", True)
         return "LEGACY-VERSION"
+
+
+def _migrate_model(model):
+    if (latest_version := _VersionManager._get_latest_version()) in _VersionManager._get_production_version():
+        if migration_fcts := MigrationConfig._get_migration_fcts_to_latest(model.version, model.config_id):
+            for fct in migration_fcts:
+                model = fct(model)
+            model.version = latest_version
```

Comment:
The model cannot be exposed to the end user, so fct cannot take a model as a parameter.